test/views/AgentAuthErrorPageViewSpec.scala

```diff
@@ -18,51 +18,49 @@ package views
 
 import org.jsoup.Jsoup
 import org.jsoup.nodes.Document
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AnyWordSpec
-import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.twirl.api.Html
 import utils.ViewTest
 
-class AgentAuthErrorPageViewSpec extends AnyWordSpec with Matchers with GuiceOneAppPerSuite with ViewTest{
+class AgentAuthErrorPageViewSpec extends ViewTest {
 
   val p1Selector = "#main-content > div > div > p:nth-child(2)"
   val p2Selector = "#main-content > div > div > p:nth-child(3)"
   val authoriseAsAnAgentLinkSelector = "#client_auth_link"
+  val anotherClientDetailsButtonSelector = "#main-content > div > div > a"
 
   val pageHeadingText = "There’s a problem"
   val pageTitleText = "There’s a problem"
   val youCannotViewText: String = "You cannot view this client’s information. Your client needs to"
   val authoriseYouAsText = "authorise you as their agent (opens in new tab)"
   val beforeYouCanTryText = "before you can sign in to this service."
-  val tryAnotherClientText = "Try another client’s details."
+  val tryAnotherClientText = "Try another client’s details"
   val authoriseAsAnAgentLink = "https://www.gov.uk/guidance/client-authorisation-an-overview"
 
   "AgentAuthErrorPageView when called in English" should {
     "render correctly" which {
-      lazy val view: Html = agentAuthErrorPageView()(fakeRequest, messages, mockConfig)
+      lazy val view: Html = agentAuthErrorPageView()(fakeRequest, messages, appConfig)
       lazy implicit val document: Document = Jsoup.parse(view.body)
 
       titleCheck(pageTitleText)
       welshToggleCheck("English")
       h1Check(pageHeadingText)
       textOnPageCheck(s"$youCannotViewText $authoriseYouAsText $beforeYouCanTryText", p1Selector)
       linkCheck(authoriseYouAsText, authoriseAsAnAgentLinkSelector, authoriseAsAnAgentLink)
-      textOnPageCheck(tryAnotherClientText, p2Selector)
+      linkCheck(tryAnotherClientText, anotherClientDetailsButtonSelector, appConfig.viewAndChangeEnterUtrUrl)
```

Comment:
English test is a "linkCheck", Welsh test below is a "buttonCheck". Should they be the same check type?

Reply:
should be buttonCheck